src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -353,6 +355,91 @@ public String addDatasetAssetSpecifications(
         .submit();
   }
 
+  public void validateDatasetAssetSpecification(Dataset dataset, AssetModel assetModel) {
```

Comment:
Do we need to call this when creating a dataset as well (right now, I think it only gets called if you're trying to add an asset to an existing dataset)

Reply:
Hmm...from what I see, it doesn't compare the asset columns and tables to the ones in the dataset (which is a big part of what your method here is doing.  You'd have to convert the datasetrequestmodel -> dataset and maybe you could call your new method from the datasetcreation validator

Reply:
It appears to perform the same three checks:
1. Asset columns match columns defined in dataset schema - [this](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java#L371) calls [isValidTableColumn](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java#L86), which refers to a map of columns defined by schema. 
2. [Valid relationship](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java#L418) defined in "follow"
3. [Validate root column and table](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java#L393)